soc: sam0: Enable generic peripheral selection

benpicco opened the pull request:
Make sure that when e.g. CONFIG_SERIAL is set, CONFIG_UART_SAM0 is
selected automatically when the sam0 SoC family is used.

This fixes several examples & tests that will otherwise only compile after manually tweaking the configuration.

galak commented:
We should open an enhancement request to cleanup the boards to remove enabling any peripherals by default that aren't needed.

ulfalizer commented:
Why don't we do e.g. this instead? Should work out to the same thing in practice, unless I'm missing something, and it centralizes SAM0 stuff.

```diff
diff --git a/drivers/serial/Kconfig.sam0 b/drivers/serial/Kconfig.sam0
index da3a5f4209..75aa63aa3c 100644
--- a/drivers/serial/Kconfig.sam0
+++ b/drivers/serial/Kconfig.sam0
@@ -5,6 +5,7 @@
 
 menuconfig UART_SAM0
        bool "Atmel SAM0 series SERCOM USART driver"
+       default y
        depends on SOC_FAMILY_SAM0
        select SERIAL_HAS_DRIVER
        select SERIAL_SUPPORT_INTERRUPT
```

ulfalizer commented:
Another option is to enable them in the board defconfig file. Might be cleanest of all.

benpicco commented:
@ulfalizer that would always enable `UART_SAM0`, no matter if `CONFIG_SERIAL` is set or not.

Maybe `depends on SOC_FAMILY_SAM0 && CONFIG_SERIAL` wold work. But I was just copying what [stm32](https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/st_stm32/common/Kconfig.defconfig.series) was doing.

ulfalizer commented:
@benpicco 
It shouldn't, because `drivers/serial/Kconfig.sam0` is `source`d within an `if SERIAL`, in `drivers/serial/Kconfig`.

benpicco committed:
soc: sam0: Enable generic peripheral selection

Make sure that when e.g. CONFIG_SERIAL is set, CONFIG_UART_SAM0 is
selected automatically when the sam0 SoC family is used.

Signed-off-by: Benjamin Valentin <benjamin.valentin@ml-pa.com>

ulfalizer reviewed (approved):
LGTM

ulfalizer commented:
@benpicco 
Can see all the dependencies on the [reference pages](https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_UART_SAM0.html) for symbols btw. Shows how the file got `source`d too.

The same information is shown in the menuconfig too, in the symbol information (press `?`).

(Trying to plug the menuconfig, because it's handy for checking changes, even though many people don't use it otherwise.)
